### DISCUSSION TITLE
Getting Git to ignore metadata and outputs in Jupyter Notebooks

### DISCUSSION THREAD

**Discussion by mmbannert** [microsoft/vscode-discussions#2416] [Extension Development QnA] [answered]
Dear VS Coders,

I frequently work with Jupyter Notebooks in VS Code. While VS Code provides good source control in principle and I am aware of the possibility [to hide changes in metadata and outputs](https://code.visualstudio.com/docs/datascience/jupyter-notebooks#_custom-notebook-diffing) when diffing two versions of a Jupyter Notebook. 

While hiding these irrelevant changes does help in diffing two Notebook versions, it is still annoying that they cause a Notebook file to show up in the "Changes" section under "Source Control".

I know of a few (inconvenient) workarounds to this problem such as [Jupytext](https://jupytext.readthedocs.io/en/latest/) or [nbstripout](https://github.com/kynan/nbstripout) but these solutions do not appear to be well-maintained. I think that VS Code should have a better solution to address this.

If there is no way to achieve this in VS Code alone, I would be curious how other users deal with the problem.

Thanks,
Michael

**Comment by isidorn** [comment, marked as answer]
I think this is a good quesiton for stack overflow.
This forum is for VS Code extension authors only.
